lms/static/js/bookmarks/views/bookmark_button.js

```diff
@@ -0,0 +1,99 @@
+;(function (define, undefined) {
+    'use strict';
+    define(['gettext', 'jquery', 'underscore', 'backbone', 'js/views/message'],
+        function (gettext, $, _, Backbone, MessageView) {
+
+        return Backbone.View.extend({
+
+            el: '.course-content',
+
+            errorIcon: '<i class="fa fa-fw fa-exclamation-triangle message-error" aria-hidden="true"></i>',
+            errorMessage: gettext('An error has occurred. Please try again.'),
+
+            srAddBookmarkText: gettext('Click to bookmark this unit'),
+            srRemoveBookmarkText: gettext('Click to remove bookmark from this unit'),
+
+            events: {
+                'click .bookmark-button': 'bookmark',
+                'sequence:changed .sequence': 'updateBookmarkState'
+            },
+
+            initialize: function () {
+                this.errorMessageView = new MessageView({
+                    el: $('.coursewide-message-banner'),
+                    templateId: '#message_banner-tpl'
+                });
+                this.bookmarksUrl = this.$('.courseware-bookmarks-button').data('bookmarksApiUrl');
+            },
+
+            bookmark: function(event) {
+                event.preventDefault();
+
+                var $buttonElement = $(event.currentTarget);
+                var usageId = $buttonElement.data("id");
+                $buttonElement.attr("disabled", true).addClass('is-disabled');
+
+                if ($buttonElement.hasClass('bookmarked')) {
+                    this.removeBookmark($buttonElement, usageId);
+                } else {
+                    this.addBookmark($buttonElement, usageId);
+                }
+            },
+
+            addBookmark: function($buttonElement, usageId) {
+                var view = this;
+                $.ajax({
+                    data: {usage_id: usageId},
+                    type: "POST",
+                    url: view.bookmarksUrl,
+                    dataType: 'json',
+                    success: function () {
+                        $('.seq-book.active').find('.bookmark-icon').removeClass('is-hidden').addClass('bookmarked');
+                        $buttonElement.removeClass('un-bookmarked').addClass('bookmarked');
+                        $buttonElement.attr('aria-pressed', 'true');
+                        $buttonElement.find('.bookmark-sr').text(view.srRemoveBookmarkText);
```

Comment:
I think we should move the logic of updating the nav icons to the sequence module javascript.